### PR TITLE
feature/SLNT-1_Initialize_extending_airbnb_config

### PR DESCRIPTION
**Pull request by MostafaFawzy7**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+module.exports = {
```
Comment (HassanElZarkawy) [inline index.js:1]: Can we put this file in a `src` directory? Keeping the conventions.

File: index.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    "plugin:prettier/recommended",
+    "plugin:@typescript-eslint/eslint-recommended",
+  ],
+  plugins: ["react", "prettier"],
+  parser: "@typescript-eslint/parser",
+  parserOptions: {
+    project: "./tsconfig.eslint.json",
+  },
+  rules: {
+    "import/no-extraneous-dependencies": "off",
```
Comment (HassanElZarkawy) [inline index.js:17]: Why did you disable this? I believe it prevents you from using dependencies that are not in `package.json`

File: package.json [hunk 3]
```diff
@@ -0,0 +1,44 @@
+{
+  "name": "eslint-config-supa",
+  "version": "1.0.1",
```
Comment (HassanElZarkawy) [inline package.json:3]: We typically use version `0.1.0` for the first release, and go on from there

**Review by HassanElZarkawy — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (MostafaFawzy7) [reply index.js]: ✅

File: index.js (hunk 2 above)
Comment (MostafaFawzy7) [reply index.js]: ✅

File: package.json (hunk 3 above)
Comment (MostafaFawzy7) [reply package.json]: ✅

**Comment by MostafaFawzy7** [comment]
> @MostafaFawzy7 There's no workflow? how are we going to integrate this?

✅

File: package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (MostafaFawzy7) [reply package.json]: ✅

File: .github/workflows/build.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+name: Node.js CI
+
+# only trigger on pull request closed events
+on:
+  pull_request:
+    types: [ closed ]
```
Comment (HassanElZarkawy) [inline .github/workflows/build.yaml:6]:
```suggestion

```

File: .github/workflows/build.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+jobs:
+  package:
+    if: github.event.pull_request.merged == true
```
Comment (HassanElZarkawy) [inline .github/workflows/build.yaml:10]:
```suggestion

```

File: .github/workflows/build.yaml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js 14.x
+        uses: actions/setup-node@v1
+        with:
+          node-version: 14.x
```
[the remark this answers is not in the document]
Comment (MostafaFawzy7) [reply .github/workflows/build.yaml]: ✅

**Comment by MostafaFawzy7** [comment]
All comments have been resolved! ✅

File: .github/workflows/build.yaml [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+jobs:
+  package:
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js 14.x
+        uses: actions/setup-node@v1
+        with:
+          node-version: latest
```
Comment (SirNarsh) [inline .github/workflows/build.yaml:18]: `latest` is not equal to `lts` , also let's switch to actions/setup-node@v3
Screenshot 2023-10-25 at 15 52 36

File: .github/workflows/build.yaml [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+    branches:
+      - main
+      - feature/SLNT-1_Initialize_extending_airbnb_config
+
+jobs:
+  package:
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js 14.x
```
Comment (SirNarsh) [inline .github/workflows/build.yaml:15]: fix name to avoid confusion

File: .github/workflows/build.yaml [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+# only trigger on pull request closed events
+on:
+  push:
+    branches:
+      - main
+      - feature/SLNT-1_Initialize_extending_airbnb_config
+
+jobs:
+  package:
+    runs-on: ubuntu-18.04
```
Comment (SirNarsh) [inline .github/workflows/build.yaml:12]: let's switch to `lts` ubuntu 20 🙏

File: .github/workflows/build.yaml [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+      - uses: actions/checkout@v2
+      - name: Use Node.js 14.x
+        uses: actions/setup-node@v1
+        with:
+          node-version: latest
+
+      - run: echo "//npm.pkg.github.com/:_authToken=${{secrets.GITHUB_TOKEN}}" >> ~/.npmrc
+      - run: npm install
+      - run: npm run build
+      - run: npm publish
```
Comment (MostafaFawzy7) [inline .github/workflows/build.yaml:23]: @SirNarsh here should we make it public?
```
- run: npm publish --access public
```

File: .github/workflows/build.yaml (hunk 8 above)
Comment (MostafaFawzy7) [reply .github/workflows/build.yaml]: ✅

File: .github/workflows/build.yaml (hunk 7 above)
Comment (MostafaFawzy7) [reply .github/workflows/build.yaml]: ✅

**Comment by MostafaFawzy7** [comment]
@SirNarsh @HassanElZarkawy Revisit again as it should be ready now!

**Review by moemen-zgtech — COMMENTED** [review]
so far looking good to me, but i'm wondering if should we add `.prettierrc` file as well ?

File: .eslintrc
[the remark this answers is not in the document]
Comment (MostafaFawzy7) [reply .eslintrc:1]: it's the same, `.eslintrc` is shortcut for `.eslintrc.json`

**Comment by MostafaFawzy7** [comment]
> so far looking good to me, but i'm wondering if should we add `.prettierrc` file as well ?

This would be added in the next ticket: `SLNT-2` which is for adding the complete config we intend to use and we already used it in `nao-referrals` app

**Review by moemen-zgtech — APPROVED** [review]

**Comment by SirNarsh** [comment]
as discussed in daily `eslint-config-supaapps` to allow sharing

https://eslint.org/docs/latest/extend/shareable-configs

**Comment by MostafaFawzy7** [comment]
@SirNarsh Check the last 3 commits 🙏

**Comment by MostafaFawzy7** [comment]
@SirNarsh Build stuff reverted!

**Review by HassanElZarkawy — APPROVED** [review]

**Review by moemen-zgtech — APPROVED** [review]

**Review by Abdallah-Abdelkhalek — APPROVED** [review]

**Pull request merged**